include/graph_generator.h

```diff
@@ -77,18 +77,16 @@ namespace detail {
 	  public:
 		/**
 		 * @param num_nodes Number of CELERITY nodes, including the master node.
-		 * @param tm
 		 * @param cdag The command graph this generator should operate on.
 		 */
-		graph_generator(size_t num_nodes, task_manager& tm, reduction_manager& rm, command_graph& cdag);
+		graph_generator(size_t num_nodes, reduction_manager& rm, command_graph& cdag);
 
 		void add_buffer(buffer_id bid, const cl::sycl::range<3>& range);
 
 		// Build the commands for a single task
-		void build_task(task_id tid, const std::vector<graph_transformer*>& transformers);
+		void build_task(const task& tsk, const std::vector<graph_transformer*>& transformers);
```

Comment:
I like the use of `task&` instead of `task*` for non-null parameters here. IMO this should be policy for all non-null `task` and `abstract_command` parameters, which are currently pointers everywhere else for some reason.

Reply:
Agreed!